pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -953,13 +969,20 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 }
 
 func setUpstreamProtocol(cluster *apiv2.Cluster, port *model.Port) {
-	if port.Protocol.IsHTTP2() {
-		cluster.Http2ProtocolOptions = &core.Http2ProtocolOptions{
-			// Envoy default value of 100 is too low for data path.
-			MaxConcurrentStreams: &types.UInt32Value{
-				Value: 1073741824,
-			},
-		}
+	if port.Protocol.IsHTTP2() && cluster.Http2ProtocolOptions != nil {
```

Comment:
I think you wanted the second check to be `cluster.Http2ProtocolOptions == nil`? But you could remove it anyway, since `setH2UpstreamProtocol()` handles both cases.